sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -89,19 +89,41 @@ RT::PiProgram ProgramManager::getBuiltOpenCLProgram(OSModuleHandle M,
   std::shared_ptr<context_impl> Ctx = getSyclObjImpl(Context);
   std::map<OSModuleHandle, RT::PiProgram> &CachedPrograms =
       Ctx->getCachedPrograms();
+
+  std::unique_lock<std::mutex> CacheLock = Ctx->lockProgramCache();
```

Comment:
Not a real problem, but I think it should be better to move this before you get the programs cache at line 90, so there is no way it can be accessed before locked. Also I was hoping to see a method that returns the cache of programs, which is already locked (as a side-effect) instead of explicit locking in the caller.

Reply:
> Also I was hoping to see a method that returns the cache of programs, which is already locked (as a side-effect) instead of explicit locking in the caller.

Done.